python/ray/_raylet.pyx

```diff
@@ -184,6 +184,96 @@ def compute_task_id(ObjectID object_id):
     return TaskID(object_id.native().TaskId().Binary())
 
 
+def setup_logging(stdout_name, stderr_name):
+    """Sets up logging for the current worker, creating the file and flushing
+    buffers as is necessary.
+
+    Args:
+        stdout_name (str): The file name that stdout should be written to.
+        stderr_name(str): The file name that stderr should be written to.
+
+    Returns:
+        (tuple) The absolute paths of the files that stdout and stderr will be
+    written to.
+    """
+    stdout_path = ""
+    stderr_path = ""
+    worker_pid = os.getpid()
+
+    cdef extern from "ray/util/util.h":
+        void flush_out()
+        void flush_err()
+
+    if stdout_name:
+        # Line-buffer the output (mode 1).
+        stdout_file = open_worker_log(stdout_name, worker_pid)
+
+        # Before redirecting stdout we need to make sure userspace buffers
+        # are all flushed.
+        # Flush python's userspace buffers
```

Comment:
I think this is a useful comment because it's important to know why we're doing an expensive operation. For example, if in the future we always start workers with `python -u` then we shouldn't have to guess if removing this line will have any other effect.

Reply:
I'm actually unclear on this⁠—what actually does happen in that case? Normally `python -u`'s job is to make `sys.stdout.flush()` unnecessary (and similar for `stderr` etc.). If the `flush()` would still be necessary in that case, it might be good to explain why in the comment? Or if it isn't, then it might be useful to leave a note that we can remove it if we switch to `python -u` (which I think we should btw)?

Reply:
I don't believe we'd have to flush if stdout was guaranteed to be unbuffered. There's also a variety of ways of making sure that stdout isn't buffered, so I'd prefer to state the rationale since we can't provide comprehensive instructions. 

Switching to an unbuffered output might make sense, but it I think it should be in it's own PR since it would have its own independent performance characteristics.

Reply:
Sounds good!